hbase-client/src/main/java/org/apache/omid/transaction/SnapshotFilterImpl.java

```diff
@@ -298,26 +295,58 @@ public CommitTimestamp locateCellCommitTimestamp(long cellStartTimestamp, long e
         return commitCache;
     }
 
-    private void buildFamilyDeletionCache(List<Cell> rawCells, Map<String, List<Cell>> familyDeletionCache) {
-
+    private void buildFamilyDeletionCache(HBaseTransaction transaction, List<Cell> rawCells, Map<String, Long> familyDeletionCache, Map<Long, Long> commitCache, Map<String,byte[]> attributeMap) throws IOException {
         for (Cell cell : rawCells) {
-            if (CellUtil.matchingQualifier(cell, CellUtils.FAMILY_DELETE_QUALIFIER) &&
-                    CellUtil.matchingValue(cell, HConstants.EMPTY_BYTE_ARRAY)) {
-
-                String row = Bytes.toString(cell.getRow());
-                List<Cell> cells = familyDeletionCache.get(row);
-                if (cells == null) {
-                    cells = new ArrayList<>();
-                    familyDeletionCache.put(row, cells);
+            if (CellUtils.isFamilyDeleteCell(cell)) {
+                String key = getRowFamilyString(cell);
+
+                if (familyDeletionCache.containsKey(key))
+                    return;
+
+                Optional<Long> commitTimeStamp = getTSIfInSnapshot(cell, transaction, commitCache);
```

Comment:
maybe first call getTSIfInTransaction because its faster and doesnt go through commit table and stuff